evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -1917,9 +1914,35 @@ def test_component_graph_does_not_define_all_edges():
         )
 
 
+def test_component_graph_defines_edges_with_bad_syntax():
+    # Graph does not define an X edge
+    with pytest.raises(
+        ValueError, match="All edges must be specified as either an input feature"
+    ):
+        ComponentGraph(
+            {
+                "Imputer": [Imputer, "X", "y"],  # offending line
+                "One Hot Encoder": [OneHotEncoder, "Imputer.x", "y"],
+                "Target Imputer": [
+                    TargetImputer,
+                    "Imputer",
+                    "One Hot Encoder.x",
+                    "y",
+                ],  # offending line: Imputer not allowed
```

Comment:
nit-pick: I think the comment would make more sense if it was above "Target Imputer"

Reply:
Going to put it next to the "Imputer" :)